lib/endpoints/boot_params.js

```diff
@@ -5,16 +5,70 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright 2020 Joyent, Inc.
  */
+var util = require('util');
 
 var restify = require('restify');
 
 var ModelServer = require('../models/server');
 var common = require('../common');
 var validation = require('../validation/endpoints');
 var ModelPlatform = require('../models/platform');
-var dns = require('dns');
+
+/*
+ * Expected boot modules format is as follows:
+ *
+ * {
+ *      "path": "boot module path to be used by booter",
+ *      "type": "Content type. Right now only base64 is supported",
+ *      "content": "File contents, if necessary encoded as specified by type"
+ * }
+ *
+ * In the future, support for new content types or additionaly members for
+ * the boot_module objects could be provided, being the defaults the originally
+ * supported values.
+ */
+function validateBootModules(boot_modules) {
+    var errs = [];
+    var i;
+    for (i = 0; i < boot_modules.length; i += 1) {
+        var bm = boot_modules[i];
+        if (!bm.path || !bm.content) {
+            errs.push(util.format(
+                'Unexpected format for boot_module: %j', bm));
+            continue;
+        }
+
+        bm.type = bm.type || 'base64';
+
+        if (bm.type !== 'base64') {
+            errs.push(util.format(
+                'Unsupported type %s for module %s', bm.type, bm.path));
+            continue;
+        }
+        if (Buffer.from(bm.content, 'base64').toString('base64') !==
```

Comment:
I'm not quite following why we need to try to convert it back into a string? Isnt' the .from check enough?
There's no utf issues with using .toString() is there?

Reply:
No, that's a way to validate if the provided content is valid `base64`. If you create a `base64` buffer from a string which isn't a proper `base64` string, and you call `toString('base64')` on that, you'll get something different from the original string.